cmd/bccatomicswap/main.go

```diff
@@ -0,0 +1,1133 @@
+		return errors.New("contract is not an atomic swap script recognized by this tool")
+	}
+
+	feePerKb, minFeePerKb, err := getFeePerKb(c)
+	if err != nil {
+		return err
+	}
+
+	refundTx, refundFee, err := buildRefund(c, cmd.contract, cmd.contractTx, feePerKb, minFeePerKb)
+	if err != nil {
+		return err
+	}
+	refundTxHash := refundTx.TxHash()
+	var buf bytes.Buffer
+	buf.Grow(refundTx.SerializeSize())
+	refundTx.Serialize(&buf)
+
+	refundFeePerKb := calcFeePerKb(refundFee, refundTx.SerializeSize())
+
+	fmt.Printf("Refund fee: %v (%0.8f BTC/kB)\n\n", refundFee, refundFeePerKb)
+	fmt.Printf("Refund transaction (%v):\n", &refundTxHash)
+	fmt.Printf("%x\n\n", buf.Bytes())
+
+	return promptPublishTx(c, refundTx, "refund")
+}
+
+func (cmd *extractSecretCmd) runCommand(c *rpc.Client) error {
+	return cmd.runOfflineCommand()
+}
+
+func (cmd *extractSecretCmd) runOfflineCommand() error {
+	// Loop over all pushed data from all inputs, searching for one that hashes
+	// to the expected hash.  By searching through all data pushes, we avoid any
+	// issues that could be caused by the initiator redeeming the participant's
+	// contract with some "nonstandard" or unrecognized transaction or script
+	// type.
+	for _, in := range cmd.redemptionTx.TxIn {
+		pushes, err := txscript.PushedData(in.SignatureScript)
+		if err != nil {
+			return err
+		}
+		for _, push := range pushes {
+			if bytes.Equal(ripemd160Hash(push), cmd.secretHash) {
+				fmt.Printf("Secret: %x\n", push)
+				return nil
+			}
+		}
+	}
+	return errors.New("transaction does not contain the secret")
+}
+
+func (cmd *auditContractCmd) runCommand(c *rpc.Client) error {
+	return cmd.runOfflineCommand()
+}
+
+func (cmd *auditContractCmd) runOfflineCommand() error {
+	contractHash160 := btcutil.Hash160(cmd.contract)
+	contractOut := -1
+	for i, out := range cmd.contractTx.TxOut {
+		sc, addrs, _, err := txscript.ExtractPkScriptAddrs(out.PkScript, chainParams)
+		if err != nil || sc != txscript.ScriptHashTy {
+			continue
+		}
+		if bytes.Equal(addrs[0].(*btcutil.AddressScriptHash).Hash160()[:], contractHash160) {
+			contractOut = i
+			break
+		}
+	}
+	if contractOut == -1 {
+		return errors.New("transaction does not contain the contract output")
+	}
+
+	pushes, err := txscript.ExtractAtomicSwapDataPushes(cmd.contract)
+	if err != nil {
+		return err
+	}
+	if pushes == nil {
+		return errors.New("contract is not an atomic swap script recognized by this tool")
+	}
+
+	contractAddr, err := btcutil.NewAddressScriptHash(cmd.contract, chainParams)
+	if err != nil {
+		return err
+	}
+	recipientAddr, err := btcutil.NewAddressPubKeyHash(pushes.RecipientHash160[:],
+		chainParams)
+	if err != nil {
+		return err
+	}
+	refundAddr, err := btcutil.NewAddressPubKeyHash(pushes.RefundHash160[:],
+		chainParams)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Contract address:        %v\n", contractAddr)
+	fmt.Printf("Contract value:          %v\n", btcutil.Amount(cmd.contractTx.TxOut[contractOut].Value))
```

Comment:
this will also reference bitcoin since the amount will be formatted with BTC at the end using the default string conversion